LabVIEW/vi.lib/MakerHub/LINX/Firmware/Source/core/device/LinxBeagleBoneBlack.cpp

```diff
@@ -526,8 +531,16 @@ LinxBeagleBoneBlack::LinxBeagleBoneBlack()
 			FILE* spiMuxHandle = fopen(m_SpiMuxPaths[i].c_str(), "r+w+");
 			if(spiMuxHandle != NULL)
 			{
-				fprintf(spiMuxHandle, "spi");
-				fclose(spiMuxHandle);							
+				// in later debian versions the state value for the clk line changed
+				if (FilePathLayout == 8)
+					fprintf(spiMuxHandle, "spi");
+				else if (FilePathLayout >= 9 && (i%3) < 2)
```

Comment:
Why the mod operator?  i only goes to 3 in the loop.

Reply:
It's just future proofing.  If we decide to add support for the 2nd SPI port then we can just add 3 more paths to the mux paths array and the same code will still work.